src/main/drivers/bus_i2c_hal.c

```diff
@@ -187,7 +187,7 @@ bool i2cWriteBuffer(I2CDevice device, uint8_t addr_, uint8_t reg_, uint8_t len_,
 
     HAL_StatusTypeDef status;
 
-    if (reg_ == 0xFF && allowRawAccess) {
+    if ((reg_ == 0xFF || len_ == 0) && allowRawAccess) {
```

Comment:
STDPeriph I2C driver probably needs patching to allow null writes as well. Can you please check?

Reply:
no need to patch "driver/bus_busdev_i2c.c" and "driver/bus_i2c_soft.c" because they are using for-loops based on paramater "len" [... for (i = 0; i < len; i++) { ...]. so it shoudl work... but how can i test it? any other i2c-drivers out there?